src/server/cluster/cluster_family.cc

```diff
@@ -906,8 +907,9 @@ void ClusterFamily::DflyMigrateAck(CmdArgList args, ConnectionContext* cntx) {
   if (!migration)
     return cntx->SendError(kIdNotFound);
 
-  // TODO add timeout for join because it can fail
```

Comment:
why/when join can fail?

Reply:
if the connection issue, the timeout for outgoing migration and new data is sent after FIN opcode (timeout is happen we start process data and send FIN opcodes and ACK one more time)